wasi/wasi.go

```diff
@@ -15,6 +15,75 @@ import (
 	"github.com/tetratelabs/wazero/wasm"
 )
 
+// Api is a documentation interface for WASI exported functions in version "wasi_snapshot_preview1"
+//
+// Note: In WebAssembly 1.0 (MVP), there may be up to one Memory per store, which means the precise memory is always
+// wasm.Store Memories index zero: `store.Memories[0].Buffer`
+//
+// See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md
+// See https://www.w3.org/TR/wasm-core-1/#memory-instances%E2%91%A0.
+type Api interface {
+	// ArgsSizesGet is a WASI function that reads command-line argument data (Args) sizes.
+	//
+	// There are two parameters and an Errno result. Both parameters are offsets in the wasm.MemoryInstance Buffer to
+	// write the corresponding sizes in uint32 little-endian encoding.
+	//
+	// * argc - is the offset to write the argument count to the wasm.MemoryInstance Buffer
+	// * argvBufSize - is the offset to write the null terminated argument length to the wasm.MemoryInstance Buffer
+	//
+	// In WebAssembly 1.0 (MVP) Text format, this signature is:
+	//	(import "wasi_snapshot_preview1" "args_sizes_get"
+	//		(func ($wasi_args_sizes_get (param $argc i32) (param $argv_buf_size i32) (result i32))
+	//
+	// For example, if Args are []string{"a","bc"} and
+	//   FunctionArgsSizesGet parameters argc=1 and argvBufSize=6, we expect `store.Memories[0].Buffer` to contain:
+	//
+	//             uint32         uint32
+	//           +--------+     +--------+
+	//           |        |     |        |
+	// []byte{?, 0, 0, 0, 2, ?, 0, 0, 0, 5, ?}
```

Comment:
Since, this is little-endian, rather it would be 
```suggestion
	// []byte{?, 2, 0, 0, 0, ?, 5, 0, 0, 0, ?}
```

Reply:
good point and yeah that's what I'm redoing the tests about now. binary encoding is not forget when asserted literally.